### PR TITLE
code review

### PR DESCRIPTION
**Pull request by miifeanyi**

### CODE REVIEW

File: .terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "zh:800e1b168149d507d23845f7a8b7e598c7dc16d2ee0f47848cf85d3e7458884f",
+    "zh:81ac3c68d6230b77740ca367e0c05a32ebb9be0fe5478c836573218a84eb3e46",
+    "zh:86536598796ba65539816f08351ac0ab32988ab84fa8f100049579996fafc800",
+    "zh:b9985c64f0f0b5bafb7067a60381fd807f7c3dd952c5d9f531385e464867bdd5",
+    "zh:c19c692896469724c6320fa7d87532ec3935e14e0e0fa0a8a0f1cf28ae7a0b0a",
+    "zh:cb8b14f246953a275ada562f5275a0d1a4938b7d20597e62fabe264012410f53",
+    "zh:cdbfa0ad87ff4d7451cfb89e53692a651d4c9cadece6845e60d986fd454b52b3",
+    "zh:ed5c4c8ae5adda37942bb15ef058c0811a95cb4c87259ae822627756dcb90efc",
+  ]
+}
```
Comment (vladimirmukhin) [inline .terraform.lock.hcl]: This file is not needed in git
Please remove it from the repository and add .terraform folder to .gitignore

File: ec2.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    network_interface_id = aws_network_interface.private-ec2-nic.id
+  }
+}
+
+#create a network interface with an IP created from the subnet created earlier
+resource "aws_network_interface" "private-ec2-nic" {
+  subnet_id       = aws_subnet.private1.id
+  private_ips     = ["192.168.128.12"]
+  security_groups = [aws_security_group.allow_web.id]
+}
```
Comment (vladimirmukhin) [inline ec2.tf]: The virtual machine is not required now, please delete it

File: sg.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  name        = "allow_web_traffic"
+  description = "Allow Web inbound traffic"
+  vpc_id      = aws_vpc.mainvpc.id
+
+  ingress {
+    description = "HTTPS Traffic"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (vladimirmukhin) [inline sg.tf]: 0.0.0.0/0 is not secure. Limit security group to your public IP only

Comment (vladimirmukhin) [reply sg.tf]: Also, it's a good idea to parametrize the IP address

File: routing-tables.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = "aws_natgateway.ngw1.id"
+
+  }
+}
+
+resource "aws_route_table" "private-route-table-2" {
+  vpc_id = "aws_vpc.mainvpc.id"
+  route {
+
```
Comment (vladimirmukhin) [inline routing-tables.tf]: This newline isn't needed

File: README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# VPC-Deployment-using-Terraform
```
Comment (vladimirmukhin) [inline README.md]: We don't yet need a readme file. Please remove it

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
1. No need to put each resource in a separate file. It's better to group them by purpose. For example vpc.tf will contain everything that's related to vpc database.tf will contain all database resources (instance, subnet group, parameter group, etc)
2. Instead of hardcoding index in the resource name, like ngw1 and ngw2 use count meta-argument:
https://www.terraform.io/docs/language/meta-arguments/count.html
3. All the name tags should contain the workspace variable. It will allow us to have unique resource names in case of multiple environments:
https://www.terraform.io/docs/language/state/workspaces.html#current-workspace-interpolation
4. There are multiple style-related mistakes. Missing newlines or newlines that are not needed. Also, I will not recommend having too many comments. The resource itself is pretty self-explanatory. You need to use comments when there are some not-standard solutions that need additional explanation Here is the PR that addresses all this:
https://github.com/miifeanyi/VPC-Deployment-using-Terraform/pull/2
Use it as a reference or just merge it to the test branch

File: .terraform.lock.hcl (hunk 1 above)
Comment (miifeanyi) [reply .terraform.lock.hcl]: Terraform dependency file removed and .terraform folder added to gitignore

File: ec2.tf (hunk 2 above)
Comment (miifeanyi) [reply ec2.tf]: virtual machine deleted

File: README.md (hunk 5 above)
Comment (miifeanyi) [reply README.md]: readme file removed

File: routing-tables.tf (hunk 4 above)
Comment (miifeanyi) [reply routing-tables.tf]: adjusted style-related mistake by referencing [#2 ](https://github.com/miifeanyi/VPC-Deployment-using-Terraform/pull/2)

**Pull request merged**